modules/documents/client/directives/documents.confirm.unpublish.directive.js

```diff
@@ -0,0 +1,302 @@
+            self.submit = submit;
+            self.cancel = cancel;
+
+            // Initialize ....
+            init();
+
+            function init() {
+              self.files = scope.files ? collect(scope.files) : [];
+              self.folders = scope.folders ? collect(scope.folders) : [];
+              self.unpublishableFiles = [];
+              self.unpublishableFolders = [];
+              // The combined list is used by the ng-repeat.
+              self.combinedList = self.folders.concat(self.files);
+              // Parse folders and determine eligibility for unpublishing.
+              parseFilesAndFolders()
+                .then(function() {
+                  updateText();
+                  self.busy = false;
+                  $scope.$apply();
+                }).catch(function (err) {
+                  self.errMsg = err.message;
+                  return reject(err);
+                });
+            }
+
+            function parseFilesAndFolders () {
+              var promises = [];
+              // Parse folders and determine eligibility for unpublishing.
+              _.forEach(self.folders, function(item) {
+                promises.push(new Promise (function (resolve, reject) {
+                  checkFolderForContent(item)
+                    .then(function() {
+                      item.canBeUnpublished = (!item.hasPublished && item.userCanUnpublish);
+                      if (item.canBeUnpublished) {
+                        self.unpublishableFolders.push(item);
+                      } else {
+                        setReasonForItem(item);
+                      }
+                      return resolve(item);
+                    }).catch(function (err) {
+                      self.errMsg = err.message;
+                      return reject(err);
+                    });
+                }));
+              });
+              // Parse files and determine eligibility for unpublishing.
+              _.forEach(self.files, function(item) {
+                promises.push(new Promise (function (resolve, reject) {
+                  checkFileForCollections(item, self.files)
+                    .then(function() {
+                      item.canBeUnpublished = (item.restrictingCollections.length < 1 && item.userCanUnpublish);
+                      if (item.canBeUnpublished) {
+                        self.unpublishableFiles.push(item);
+                      } else {
+                        setReasonForItem(item);
+                      }
+                      return resolve(item);
+                    }).catch(function (err) {
+                      self.errMsg = err.message;
+                      return reject(err);
+                    });
+                }));
+              });
+              return Promise.all(promises);
+            }
+
+            // Collect unpublishable folders and files.
+            function collect(items) {
+              // Filter out items that are already unpublished
```

Comment:
This comment seems out of context here. Maybe it should be moved outside of the function, as an addition to what is on line 110?

Reply:
Ah, this was supposed to be removed. Thanks for the reminder, will update.